datasource/threedi_results.py

```diff
@@ -420,10 +341,101 @@ def ds_aggregation(self):
 
         # Load aggregation netcdf
         try:
-            aggregation_netcdf_file = find_aggregation_netcdf_gw(self.file_path)
+            aggregation_netcdf_file = find_aggregation_netcdf(self.file_path)
         except IndexError:
             logger.error("Could not find the aggregation netcdf.")
             return None
         else:
             logger.info("Opening aggregation netcdf: %s" % aggregation_netcdf_file)
             return h5py.File(aggregation_netcdf_file, mode="r")
+
+
+def normalized_object_type(current_layer_name):
+    """Get a normalized object type for internal purposes."""
+    if current_layer_name in list(layer_object_type_mapping.keys()):
+        return layer_object_type_mapping[current_layer_name]
+    else:
+        msg = "Unsupported layer: %s." % current_layer_name
+        logger.warning(msg)
+        return None
+
+
+def find_h5_file(netcdf_file_path):
+    """An ad-hoc way to get the h5_file.
+
+    We assume the h5_file file is in on of the following locations (note:
+    this order is also the searching order):
+
+    1) . (in the same dir as the netcdf)
+    2) ../preprocessed
+
+    relative to the netcdf file and has extension '.h5'
+
+    Args:
+        netcdf_file_path: path to the result netcdf
+
+    Returns:
+        h5_file path
+
+    Raises:
+        IndexError if nothing is found
+    """
+    pattern = "*.h5"
+    inpdir = os.path.join(os.path.dirname(netcdf_file_path), "..", "preprocessed")
+    resultdir = os.path.dirname(netcdf_file_path)
+
```

Comment:
Zoiets lijkt me netter:

    for directory in [inpdir, resultdir]:
        h5_files = glob.glob(..._
        if h5_files:
            return h5files[0]
    raise FileNotFound(zoiets, even kijken welke error het precies is)

Eventueel een eigen kleine Exception voor aanmaken. H5NotFound. Kan ook.